sds/src/scanner/mod.rs

```diff
@@ -1718,6 +1648,175 @@ mod test {
         assert_eq!(matches.len(), 1);
     }
 
+    #[test]
+    fn test_included_keyword_match_further_than_look_ahead_character_count() {
+        let redact_test_rule = RegexRuleConfig::builder("world".to_owned())
+            .match_action(MatchAction::Redact {
+                replacement: "[REDACTED]".to_string(),
+            })
+            .proximity_keywords(ProximityKeywordsConfig {
+                look_ahead_character_count: 30,
+                included_keywords: vec!["hello".to_string()],
+                excluded_keywords: vec![],
+            })
+            .build();
+
+        let scanner = ScannerBuilder::new(&[redact_test_rule])
+            .with_keywords_should_match_event_paths(true)
+            .build()
+            .unwrap();
+
+        let mut content = "hello [this block is exactly 37 chars long] world".to_string();
+        let matches = scanner.scan(&mut content);
+
+        // There should be no matches because the included keyword is too far from the actual match
+        assert_eq!(matches.len(), 0);
+    }
```

Comment:
I think it's a really good example of descriptive test:
- the name is clear
- `"hello [this block is exactly 37 chars long] world"` is super explicit 💯 

a small nipick:
you can possibly drop the comment by changing the test name:
test_included_keyword_**not**_match_further_than_look_ahead_character_count